server/catgenome/src/main/java/com/epam/catgenome/client/rscb/RCSBApiBuilder.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2021 EPAM Systems
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.epam.catgenome.client.rscb;
+
+import com.epam.catgenome.controller.JsonMapper;
+import com.epam.catgenome.util.QueryUtils;
+import lombok.RequiredArgsConstructor;
+import okhttp3.OkHttpClient;
+import retrofit2.Retrofit;
+import retrofit2.converter.jackson.JacksonConverterFactory;
+
+import java.util.concurrent.TimeUnit;
+
+@RequiredArgsConstructor
+public class RCSBApiBuilder {
+    private static final JacksonConverterFactory CONVERTER =
+            JacksonConverterFactory
+                    .create(new JsonMapper());
+
+    private final int connectTimeout;
+    private final int readTimeout;
+    private final String apiHost;
+
+    public RCSBApi buildClient() {
+        return new Retrofit.Builder()
+                .baseUrl(QueryUtils.normalizeUrl(apiHost))
+                .addConverterFactory(CONVERTER)
+                .client(buildHttpClient())
+                .build()
+                .create(RCSBApi.class);
+    }
+
+    private OkHttpClient buildHttpClient() {
+        OkHttpClient.Builder builder = new OkHttpClient.Builder();
```

Comment:
Now it could be rewritten without intermediate links, since `hostnameVerifier` method also returns `builder`

Reply:
fixed!